### ISSUE TITLE
If a text is a child of an object, NVDA fails to read the last character of the text

### ISSUE THREAD

**Issue by k-kolev1985** [nvaccess/nvda#8093]
### Steps to reproduce:
1. Open Windows Protection (I think that is how it is called in english, or "Windows Security").
2. Use TAB to go to the "Items" list.
3. Use the arrow keys to select "Viruses and threats protection" and open it with ENTER.
4. The focus should land on the "Quick Scan" button. If it does not, use TAB/Shift+TAB to move the focus to it.
5. Go down 1 level in the objects hierarchy for the button to the text "Quick Scan".
6. Read the "Quick Scan" text with the commands for reading character by character in the current text object.

### Expected behavior:
NVDA should read the whole text - from the first to the last character included.

### Actual behavior:
NVDA fails to read the last character of the text string. It thinks that the character before it is the last one.

### System configuration:
- Operating system: Windows 10 Pro version 1809 (build 17763.475), 64-bit, in Bulgarian with all locale settings set to "Bulgarian".
- NVDA version: alpha-17148,029c0e31, installed, in bulgarian.
- Processor: Intel Core i5-4670 at 3.40GHz.
- RAM Memory: 8.00GB.
- Graphics: MSI GeForce GTX-1050TI Gaming X 4G, 4096MB dedicated memory + 4065MB shared system memory, desktop resolution set to 1920x1080.
- Sound Card: Realtek ALC887 at Intel Lynx Point PCH - High Definition Audio Controller.

### Additional information:
1. I'm using Windows Security/Protection in the case to demonstrate the issue, but I think it occurs in all similar cases in which the text object is a child of another object.

**Comment by leonardder** [comment]
cc @michaeldcurran @josephsl

**Comment by derekriemer** [comment]
Confirmed on 1709

**Comment by Adriani90** [comment]
@k-kolev1985 are you still seeing this issue? I am testing in last Skype UWP but I cannot reproduce it with NVDA 2019.1.1 on Windows 10 1809 Update.

**Comment by k-kolev1985** [comment]
@Adriani90 Yes, it is no longer reproducible with Skype UWP. But that is mainly due to changes in the app itself. Because I've managed to reproduce it someplace else - in the button to make a fast check for viruses in Windows Defender. I've gave Skype UWP as an easy to reproduce example, but apparently the issue is not restricted only to it.

**Comment by Adriani90** [comment]
Thanks for reporting back. Maybe if you have time, and you come accross this issue in an app, please post steps to reproduce to be able to track this issue. Without any test case it will be quite hard to solve this.
Can the issue in Windows defender be reproduce reliably, Or is it occuring only in certain cases?

**Comment by k-kolev1985** [comment]
@Adriani90 I've updated the main comment/report on the top to reflect the new test case with Windows Security/Protection.

P.S.: I'm not sure if it is "Windows Security" or "Windows Protection" in english, because my main system is in bulgarian and right now I'm too lazy (it is late evening here) to power on the virtual machine which is in english to check it out. sorry...

**Comment by Adriani90** [comment]
This is reproducible in windows security on Windows 10 1909 Update.

**Comment by tmthywynn8** [comment]
I can still reproduce this with NVDA 2022.4 and Windows 11 22H2.

**Comment by Adriani90** [comment]
Closing in favor of #13302 which contains some more discussion.
